haproxy_discovery.sh

```diff
@@ -14,6 +14,10 @@
 HAPROXY_SOCK="/var/run/haproxy/info.sock"
 [ -n "$1" ] && echo $1 | grep -q ^/ && HAPROXY_SOCK="$(echo $1 | tr -d '\040\011\012\015')"
 
+# Define variables and perform checks
+CONFIG_FILE="$(dirname $(readlink -f "$0"))/haproxy_config"
+. $CONFIG_FILE
+
```

Comment:
config file can things more manageable, yet, I strongly believe, that script should be able to run on it's own, using environment variables and fallback defaults, and source the config file, if it exists.
thus, I propose to update both scripts with sane defaults, i.e.:
```
# haproxy_discovery.sh
CONFIG_FILE="${CONFIG_FILE:-$(dirname $(readlink -f "$0"))/zabbix-haproxy.conf}"
[ -r "$CONFIG_FILE" ] && source $CONFIG_FILE
[ -n "$1" ] && echo $1 | grep -q ^/ && HAPROXY_SOCKET="$(echo $1 | tr -d '\040\011\012\015')"
HAPROXY_SOCKET="${HAPROXY_SOCKET:-/var/run/haproxy/info.sock}"
```
In both scripts, command line HAPROXY_SOCKET should take precedence, if passed as argument. Otherwise it should attempt to read it from config. Failing that, it should use the default.
```
# haproxy_stats.sh
set -o pipefail
CONFIG_FILE="${CONFIG_FILE:-$(dirname $(readlink -f "$0"))/zabbix-haproxy.conf}"
[ -r "$CONFIG_FILE" ] && source $CONFIG_FILE
if [[ "$1" = /* ]]
then
  HAPROXY_SOCKET="$1"
  shift 1
fi
HAPROXY_SOCKET="${HAPROXY_SOCKET:-/var/run/haproxy/info.sock}"
```
and yes, both scripts should use the same variable name for haproxy socket :)